### DISCUSSION TITLE
masterproc output to cesm.log?

### DISCUSSION THREAD

**Discussion by samsrabin** [ESCOMP/CTSM#1840] [Q&A] [answered]
In some troubleshooting, I wanted to just output one processor's worth of messages to the cesm.log file. That's where the rest of my troubleshooting messages are going, and I want to be able to view all the messages in one file to get them in chronological order. However, `if (masterproc) then write(iulog, ...` sends the message to lnd.log. Is there any way to redirect the master processor's output—ideally for just some messages—to lnd.log? Or alternatively, can I redirect an arbitrary processor's messages to cesm.log?

**Comment by ekluzek** [comment, marked as answer]
Output from any processor other than masterproc will go to cesm.log. While output on masterproc goes to the lnd.log. Adding the if masterproc just suppresses the output from other processors.

The redirection to each file happens in the NUOPC cap for CTSM. So for your own experiment you should be able to change it so that all output goes to only one of the files for all processors. That would be a temporary hack to add to your own code.

The other way to do it would be to send the output you want from other processors to masterproc and then output it all on on masterproc.

**Reply by samsrabin** [reply]
That second option sounds like it has potential. How do I do that?

**Reply by ekluzek** [reply]
An MPI gather call. I would recommend using the shr_mpi_mod.F90 wrapper thats in components/cdeps/share with the shr_mpi_gathScatVInit and shr_mpi_gatherV wrappers. I think that will help you to see how to get it setup.
